### PR TITLE
Closes #2238: clarify dates values in flag_imputation argument of derive_vars…

### LINKED ISSUE #2238: Documentation: Clarify `dates` option in `flag_imputation` argument for `derive_vars_dt()`
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

image

# Definition of Done

- [ ] Documentation is update to clarify argument options 
- [ ] Verify if updates to other Date/Time functions need updates
- [ ] Review vignettes and templates to see if updates are needed.

### PR DESCRIPTION
**Pull request by PoojaKumari05**
…_dt() function

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_date_vars.R [hunk 1]
```diff
@@ -1263,10 +1263,15 @@ compute_tmf <- function(dtc,
 #'
 #' @param flag_imputation Whether the date imputation flag must also be derived.
 #'
-#'   If `"auto"` is specified, the date imputation flag is derived if the
-#'   `date_imputation` argument is not null.
+#'   If `"auto"` is specified and `date_imputation` argument is not null or ,
+#'   then date imputation flag is derived.
```
Comment (bundfussr) [inline R/derive_date_vars.R:1267]:
```suggestion
#'   If `"auto"` is specified and `highest_imputation` argument is not `"n"`,
#'   then date imputation flag is derived.
```
I think we missed to update this when we introduced the `highest_imputation` argument.

Could you update the description in `derive_vars_dtm()` as well?

File: R/derive_date_vars.R [hunk 2]
```diff
@@ -1263,10 +1263,15 @@ compute_tmf <- function(dtc,
...
+#'
+#'   If `"date"` is specified, then date imputation flag is derived.
+#'
+#'   If `"none"` is specified, then no date imputation flag is derived.
 #'
 #'   *Permitted Values*: `"auto"`, `"date"` or `"none"`
+#'   *Default*: `"auto"`
```
Comment (bundfussr) [inline R/derive_date_vars.R:1274]: Should be removed because we decided that the default value should not be stated in the argument description.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: R/derive_date_vars.R (hunk 1 above)
Comment (PoojaKumari05) [reply R/derive_date_vars.R]: updated as suggested

**Comment by bms63** [comment]
@bundfussr I think Pooja made all your suggested updates.  This looks good to me  - okay to merge?

**Review by bundfussr — APPROVED** [review]

**Comment by bundfussr** [comment]
> @bundfussr I think Pooja made all your suggested updates. This looks good to me - okay to merge?

@bms63 , yes, go ahead.

**Pull request merged**